core/src/main/java/org/pentaho/platform/util/EncryptedPasswordAuthenticator.java

```diff
@@ -0,0 +1,67 @@
+/*!
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ *
+ * Copyright (c) 2002-2023 Hitachi Vantara. All rights reserved.
+ *
+ */
+
+package org.pentaho.platform.util;
+
+import javax.mail.Authenticator;
+import javax.mail.PasswordAuthentication;
+
+/**
+ * Password Authenticator for Pentaho stored encrypted passwords.
+ */
+public class EncryptedPasswordAuthenticator extends Authenticator {
+
+  public EncryptedPasswordAuthenticator( String userId, String password, Base64PasswordService base64PasswordService  ) {
+    this.userId = userId;
+    this.password = password;
+    this.base64PasswordService = base64PasswordService;
+  }
+
+  public EncryptedPasswordAuthenticator( String userId, String password ) {
+    this( userId, password, new Base64PasswordService() );
+  }
+
+  final String password;
+  final String userId;
+  // Using this class to be consistent with previous implementations. Provides basic encryption/decryption base64
+  final Base64PasswordService base64PasswordService;
+
+  public static final String PASSWORD_PREFIX = "ENC:"; // standard PUC prefix
```

Comment:
getting sonar qube warning on this: https://sonar.orl.eng.hitachivantara.com/security_hotspots?id=pentaho%3Apentaho-platform-ce-parent&pullRequest=5185&hotspots=AYXpdAy3QTi5PMYJZY46

I think, it is just picking up the substring "password"

My mistake. I'd just change it to "PREFIX" or "ENCRYPTED_PREFIX"